File: LeetCode/shuffle_the_array/shuffle_the_array.c
int* shuffle(int* nums, int numsSize, int n, int* returnSize){
    //Make array
    int *shuffled_array = (int *) calloc(2*n, sizeof(int));   
    
    //Get the size of the array
    int arr_size = 2*n;
    *returnSize = 2*n;
    //this will begin our y values
    
    int x = 0;
    int y = n;
    
    //At most O(n)
    for (int i = 0; i < arr_size; i +=1) {
        
        shuffled_array[i] = nums[x];
        //printf("%d ", shuffled_array[i]);
        x +=1;
        i +=1; //Go to next position
        shuffled_array[i] = nums[y];
        y +=1;
        //printf("%d ", shuffled_array[i]);
    }
    return shuffled_array;
    
}

File: LeetCode/shuffle_the_array/shuffle_the_array.py
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        arr = []
        i = 0
        x = 0
        y = n
        
        while (i < 2*n):
            arr.insert(i, nums[x])
            i +=1
            x +=1
            arr.insert(i, nums[y])
            y +=1
            i +=1
            
        return arr
        
        